src/machine/machine_atsamd51.go

```diff
@@ -2299,3 +2299,81 @@ func checkFlashError() error {
 
 	return nil
 }
+
+// Watchdog provides access to the hardware watchdog available
+// in the SAMD51.
+var Watchdog = &watchdogImpl{}
+
+const (
+	// WatchdogMaxSources is 8
+	//
+	// SAMD51 only supports a single source in hardware, so
+	// the support for multiple sources is in software.
+	WatchdogMaxSources = 8
```

Comment:
I just noticed this value is a result of the software implementation, not the actual hardware. Should we really provide a software abstraction over the hardware at this level? 

Maybe this is worth implementing separately as a WatchdogMultiplexer type, within or out of `machine` package that receives a `Watcher` interface.

Reply:
> Should we really provide a software abstraction over the hardware at this level?

That is exactly the question, and I think the answer is probably no.